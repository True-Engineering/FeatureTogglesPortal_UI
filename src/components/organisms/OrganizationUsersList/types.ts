import { IFiltersPaneSearchPayload } from '@true-engineering/true-react-common-ui-kit';

// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
export type IFilterValuesConfig = {
  projectName: string;
};
export type IFilterValues = Partial<IFilterValuesConfig>;

// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
export type ISearchValuesConfig = {
  userName: string;
};
export type ISearchValues = IFiltersPaneSearchPayload<
  keyof ISearchValuesConfig
>;
