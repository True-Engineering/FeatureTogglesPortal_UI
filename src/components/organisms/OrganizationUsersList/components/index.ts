export * from './MemberAccordionTitle';
export * from './AddToProjectButton';
