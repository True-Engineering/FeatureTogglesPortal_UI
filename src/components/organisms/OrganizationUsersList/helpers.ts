import { uniqArray } from '../../../utils';
import { IOrganizationMember } from '../../../types';
import { IFilterValues, ISearchValues } from './types';

export const getProjectNameOptions = (members: IOrganizationMember[]) =>
  uniqArray(
    members
      .map(({ projects }) =>
        projects.map(({ projectName }) => projectName).flat(),
      )
      .flat(),
  );

export const filterMembers = ({
  members,
  searchValue,
  filterValues,
}: {
  members: IOrganizationMember[];
  searchValue?: ISearchValues;
  filterValues?: IFilterValues;
}): IOrganizationMember[] => {
  const filterProjectNameValue = filterValues?.projectName?.toUpperCase();
  const filteredByProjectName =
    filterProjectNameValue === undefined
      ? members
      : members.filter(({ projects }) =>
          projects.some(
            ({ projectName }) =>
              projectName.toUpperCase() === filterProjectNameValue,
          ),
        );

  const searchUserNameValue = searchValue?.value?.toUpperCase();
  const filteredByUserName =
    searchUserNameValue === undefined
      ? filteredByProjectName
      : filteredByProjectName.filter(({ user }) =>
          user.userName.toUpperCase().includes(searchUserNameValue),
        );

  return filteredByUserName;
};
