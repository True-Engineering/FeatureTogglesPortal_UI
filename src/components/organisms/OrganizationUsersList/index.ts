export { default as OrganizationUsersList } from './OrganizationUsersList';
export * from './OrganizationUsersList';

export type { OrganizationUsersListStyles } from './OrganizationUsersList.styles';
