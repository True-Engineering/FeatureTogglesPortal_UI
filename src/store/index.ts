import Store from './store';

const store = new Store();
export default store;

export * from './store';
export * from './context';
