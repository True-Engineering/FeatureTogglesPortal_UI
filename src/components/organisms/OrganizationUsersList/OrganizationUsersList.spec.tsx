/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react';
import { fireEvent, render, waitFor, within } from '@testing-library/react';
import { nullToUndef } from '@true-engineering/true-react-api-client';
import { clickTextInPopper, customRender } from '../../../utils';
import { apiClient } from '../../../transport';
import store, { Context } from '../../../store';
import * as fixtures from '../../../transport/mocks/fixtures';
import OrganizationUsersList from './OrganizationUsersList';

beforeEach(() => apiClient.clearLogs());

const organizations = nullToUndef(fixtures.organizations);
const members = nullToUndef(fixtures.organizationMembers);

apiClient.setMocks({
  requestName: 'getOrganizationMembers',
  fixture: members,
});

apiClient.setMocks({
  requestName: 'loadOrganizations',
  fixture: organizations,
});

apiClient.setMocks({
  requestName: 'deleteOrganizationMember',
  fixture: true,
});

describe('OrganizationUsersList рендерится', () => {
  test('Рендер на мок данных', async () => {
    const { container } = await render(
      <Context.Provider value={store}>
        <OrganizationUsersList organizationId={0} projects={[]} />
      </Context.Provider>,
    );

    await waitFor(() => apiClient.getOrganizationMembers);

    expect(apiClient.getLoggedRequests()).toContainEqual(
      expect.objectContaining({
        name: 'getOrganizationMembers',
      }),
    );

    expect(container).toMatchSnapshot();
  });
});

describe('OrganizationUsersList корректно отрабатывает сценарии', () => {
  test('Корректно фильтруется по имени', async () => {
    const { getByTestId, container } = render(
      <Context.Provider value={store}>
        <OrganizationUsersList organizationId={0} projects={[]} />
      </Context.Provider>,
    );

    await waitFor(() => apiClient.getOrganizationMembers);

    const input = getByTestId('filterspane-search-input-input');
    fireEvent.change(input, { target: { value: 'се' } });

    const names = Array.from(
      container.querySelectorAll('[class^="userName"]'),
    ).map(cell => cell.innerHTML);
    const expected = ['Семен Гусев', 'Сергей Антонов'];

    expect(names).toEqual(expected);
  });

  test('Корректно фильтруется по проекту', async () => {
    const { getByTestId, container } = await customRender(
      <OrganizationUsersList organizationId={0} projects={[]} />,
    );

    await waitFor(() => apiClient.getOrganizationMembers);

    const select = getByTestId('filterspane-filter-projectName');
    fireEvent.click(select.childNodes[0]);

    const utils = within(select);
    const option = await utils.getByText('NBA');
    fireEvent.click(option);

    const expected = ['Кент Кларк', 'Сергей Антонов'];
    const names = Array.from(
      container.querySelectorAll('[class^="userName"]'),
    ).map(cell => cell.innerHTML);

    expect(names).toEqual(expected);
  });

  // Тест падает с ошибкой в пайплайне и на некоторых версиях библиотек (не выявлено на каких)
  test.skip('Отправляет запрос на удаление по нажатию на кнопку удалить', async () => {
    const { container } = await customRender(
      <OrganizationUsersList organizationId={0} isEditable projects={[]} />,
    );

    await waitFor(() => apiClient.getOrganizationMembers);

    const row = container.querySelectorAll('[class^="header"]')[0];
    const utils = within(row as HTMLElement);
    const removeButton = utils.getByRole('button');
    await clickTextInPopper(removeButton, 'Удалить');

    await waitFor(() => apiClient.deleteOrganizationMember);

    const lastLogRequest = apiClient.getLoggedRequests().slice(-1)[0];
    expect(lastLogRequest.name).toBe('deleteOrganizationMember');
  });

  test('Корректная работа кнопки "Добавить проект"', async () => {
    const { getByText } = await customRender(
      <OrganizationUsersList
        organizationId={0}
        projects={organizations[0].projects}
        isEditable
      />,
    );

    await waitFor(() => apiClient.getOrganizationMembers);

    const name = await getByText('Сергей Антонов');
    fireEvent.click(name);

    const addButton = await getByText('Добавить проект');
    await clickTextInPopper(addButton, 'Cool project');

    await waitFor(() => apiClient.editMemberProjectPermission);

    const lastLogRequest = apiClient.getLoggedRequests().slice(-1)[0];
    expect(lastLogRequest.name).toBe('editMemberProjectPermission');
  });
});
