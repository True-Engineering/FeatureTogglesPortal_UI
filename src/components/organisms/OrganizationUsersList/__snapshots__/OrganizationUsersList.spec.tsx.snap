// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationUsersList рендерится Рендер на мок данных 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="filtersPane"
    >
      <div
        class="root"
        data-testid="filterspane"
      >
        <div
          class="root"
          data-testid="filterspane-search"
        >
          <div
            class="root"
            data-testid="filterspane-search-input"
          >
            <div
              class="icon"
            >
              <div
                class="root"
              >
                <svg
                  fill="none"
                  height="100%"
                  viewBox="0 0 20 20"
                  width="100%"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M9 2c3.866 0 7 3.134 7 7 0 1.7-.606 3.259-1.614 4.472l2.589 2.59c.39.39.39 1.023 0 1.414-.39.39-1.024.39-1.414 0l-2.662-2.661C11.785 15.563 10.443 16 9 16c-3.866 0-7-3.134-7-7s3.134-7 7-7zm0 2C6.239 4 4 6.239 4 9s2.239 5 5 5 5-2.239 5-5-2.239-5-5-5z"
                    fill="currentColor"
                    fill-rule="evenodd"
                    stroke="none"
                  />
                </svg>
              </div>
            </div>
            <div
              class="root"
            >
              <div
                class="inputWrapper"
              >
                <input
                  class="input"
                  data-testid="filterspane-search-input-input"
                  type="text"
                  value=""
                />
                <span
                  class="label"
                >
                  Поиск
                </span>
              </div>
            </div>
          </div>
          <div
            class="selectWrapper"
          />
        </div>
        <div
          class="filterWithClearButton"
        >
          <div
            class="root noValue"
            data-testid="filterspane-filter-projectName"
          >
            <div
              class="item"
            >
              <div
                class="name"
              >
                Проект
              </div>
              <div
                class="iconContainer"
              >
                <div
                  class="root"
                >
                  <svg
                    fill="none"
                    height="100%"
                    viewBox="0 0 20 20"
                    width="100%"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M10.0029 11.6053L5.71004 7.31242C5.31951 6.9219 4.68635 6.9219 4.29582 7.31242C3.9053 7.70295 3.9053 8.33611 4.29582 8.72664L9.29582 13.7266C9.68635 14.1172 10.3195 14.1172 10.71 13.7266L15.71 8.72664C16.1006 8.33611 16.1006 7.70295 15.71 7.31242C15.3195 6.9219 14.6863 6.9219 14.2958 7.31242L10.0029 11.6053Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                      stroke="none"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            class="clear"
          >
            <button
              class="root s light"
              data-testid="filterspane-clear-button"
              type="button"
            >
              <span
                class="content"
              >
                Очистить
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="root"
    >
      <div
        class="item"
      >
        <div
          class="header"
        >
          <div
            class="arrow"
          >
            <div
              class="root"
            >
              <svg
                fill="none"
                height="100%"
                viewBox="0 0 20 20"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M11.5858 10L7.29289 5.70711C6.90237 5.31658 6.90237 4.68342 7.29289 4.29289C7.68342 3.90237 8.31658 3.90237 8.70711 4.29289L13.7071 9.29289C14.0976 9.68342 14.0976 10.3166 13.7071 10.7071L8.70711 15.7071C8.31658 16.0976 7.68342 16.0976 7.29289 15.7071C6.90237 15.3166 6.90237 14.6834 7.29289 14.2929L11.5858 10Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  stroke="none"
                />
              </svg>
            </div>
          </div>
          <div
            class="title"
          >
            <div
              class="root"
            >
              <div
                class="userInfo"
              >
                <img
                  alt="user-avatar"
                  class="imageAvatar"
                  src="https://www.gravatar.com/avatar/67868768758.jpg?d=identicon"
                />
                <div
                  class="userName"
                  title="Иванов Иван"
                >
                  Иванов Иван
                </div>
              </div>
              <div
                class="projects"
              >
                <div
                  class="tag"
                >
                  Smart
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="header"
        >
          <div
            class="arrow"
          >
            <div
              class="root"
            >
              <svg
                fill="none"
                height="100%"
                viewBox="0 0 20 20"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M11.5858 10L7.29289 5.70711C6.90237 5.31658 6.90237 4.68342 7.29289 4.29289C7.68342 3.90237 8.31658 3.90237 8.70711 4.29289L13.7071 9.29289C14.0976 9.68342 14.0976 10.3166 13.7071 10.7071L8.70711 15.7071C8.31658 16.0976 7.68342 16.0976 7.29289 15.7071C6.90237 15.3166 6.90237 14.6834 7.29289 14.2929L11.5858 10Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  stroke="none"
                />
              </svg>
            </div>
          </div>
          <div
            class="title"
          >
            <div
              class="root"
            >
              <div
                class="userInfo"
              >
                <img
                  alt="user-avatar"
                  class="imageAvatar"
                  src="https://www.gravatar.com/avatar/678687683358.jpg?d=identicon"
                />
                <div
                  class="userName"
                  title="Петров Василий"
                >
                  Петров Василий
                </div>
              </div>
              <div
                class="projects"
              >
                <div
                  class="tag"
                >
                  Smart
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="header"
        >
          <div
            class="arrow"
          >
            <div
              class="root"
            >
              <svg
                fill="none"
                height="100%"
                viewBox="0 0 20 20"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M11.5858 10L7.29289 5.70711C6.90237 5.31658 6.90237 4.68342 7.29289 4.29289C7.68342 3.90237 8.31658 3.90237 8.70711 4.29289L13.7071 9.29289C14.0976 9.68342 14.0976 10.3166 13.7071 10.7071L8.70711 15.7071C8.31658 16.0976 7.68342 16.0976 7.29289 15.7071C6.90237 15.3166 6.90237 14.6834 7.29289 14.2929L11.5858 10Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  stroke="none"
                />
              </svg>
            </div>
          </div>
          <div
            class="title"
          >
            <div
              class="root"
            >
              <div
                class="userInfo"
              >
                <div
                  class="wordAvatar orange-yellow"
                >
                  КК
                </div>
                <div
                  class="userName"
                  title="Кент Кларк"
                >
                  Кент Кларк
                </div>
              </div>
              <div
                class="projects"
              >
                <div
                  class="tag"
                >
                  Smart
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="header"
        >
          <div
            class="arrow"
          >
            <div
              class="root"
            >
              <svg
                fill="none"
                height="100%"
                viewBox="0 0 20 20"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M11.5858 10L7.29289 5.70711C6.90237 5.31658 6.90237 4.68342 7.29289 4.29289C7.68342 3.90237 8.31658 3.90237 8.70711 4.29289L13.7071 9.29289C14.0976 9.68342 14.0976 10.3166 13.7071 10.7071L8.70711 15.7071C8.31658 16.0976 7.68342 16.0976 7.29289 15.7071C6.90237 15.3166 6.90237 14.6834 7.29289 14.2929L11.5858 10Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  stroke="none"
                />
              </svg>
            </div>
          </div>
          <div
            class="title"
          >
            <div
              class="root"
            >
              <div
                class="userInfo"
              >
                <img
                  alt="user-avatar"
                  class="imageAvatar"
                  src="https://www.gravatar.com/avatar/67868768758.jpg?d=identicon"
                />
                <div
                  class="userName"
                  title="Валерия Климова"
                >
                  Валерия Климова
                </div>
              </div>
              <div
                class="projects"
              >
                <div
                  class="tag"
                >
                  Smart
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="header"
        >
          <div
            class="arrow"
          >
            <div
              class="root"
            >
              <svg
                fill="none"
                height="100%"
                viewBox="0 0 20 20"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M11.5858 10L7.29289 5.70711C6.90237 5.31658 6.90237 4.68342 7.29289 4.29289C7.68342 3.90237 8.31658 3.90237 8.70711 4.29289L13.7071 9.29289C14.0976 9.68342 14.0976 10.3166 13.7071 10.7071L8.70711 15.7071C8.31658 16.0976 7.68342 16.0976 7.29289 15.7071C6.90237 15.3166 6.90237 14.6834 7.29289 14.2929L11.5858 10Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  stroke="none"
                />
              </svg>
            </div>
          </div>
          <div
            class="title"
          >
            <div
              class="root"
            >
              <div
                class="userInfo"
              >
                <div
                  class="wordAvatar orange-yellow"
                >
                  ПВ
                </div>
                <div
                  class="userName"
                  title="Петров Василий"
                >
                  Петров Василий
                </div>
              </div>
              <div
                class="projects"
              >
                <div
                  class="tag"
                >
                  Smart
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="header"
        >
          <div
            class="arrow"
          >
            <div
              class="root"
            >
              <svg
                fill="none"
                height="100%"
                viewBox="0 0 20 20"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M11.5858 10L7.29289 5.70711C6.90237 5.31658 6.90237 4.68342 7.29289 4.29289C7.68342 3.90237 8.31658 3.90237 8.70711 4.29289L13.7071 9.29289C14.0976 9.68342 14.0976 10.3166 13.7071 10.7071L8.70711 15.7071C8.31658 16.0976 7.68342 16.0976 7.29289 15.7071C6.90237 15.3166 6.90237 14.6834 7.29289 14.2929L11.5858 10Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  stroke="none"
                />
              </svg>
            </div>
          </div>
          <div
            class="title"
          >
            <div
              class="root"
            >
              <div
                class="userInfo"
              >
                <div
                  class="wordAvatar orange-green"
                >
                  СГ
                </div>
                <div
                  class="userName"
                  title="Семен Гусев"
                >
                  Семен Гусев
                </div>
              </div>
              <div
                class="projects"
              >
                <div
                  class="tag"
                >
                  Smart
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="header"
        >
          <div
            class="arrow"
          >
            <div
              class="root"
            >
              <svg
                fill="none"
                height="100%"
                viewBox="0 0 20 20"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M11.5858 10L7.29289 5.70711C6.90237 5.31658 6.90237 4.68342 7.29289 4.29289C7.68342 3.90237 8.31658 3.90237 8.70711 4.29289L13.7071 9.29289C14.0976 9.68342 14.0976 10.3166 13.7071 10.7071L8.70711 15.7071C8.31658 16.0976 7.68342 16.0976 7.29289 15.7071C6.90237 15.3166 6.90237 14.6834 7.29289 14.2929L11.5858 10Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  stroke="none"
                />
              </svg>
            </div>
          </div>
          <div
            class="title"
          >
            <div
              class="root"
            >
              <div
                class="userInfo"
              >
                <div
                  class="wordAvatar orange-yellow"
                >
                  ПВ
                </div>
                <div
                  class="userName"
                  title="Петров Василий"
                >
                  Петров Василий
                </div>
              </div>
              <div
                class="projects"
              >
                <div
                  class="tag"
                >
                  Smart
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="header"
        >
          <div
            class="arrow"
          >
            <div
              class="root"
            >
              <svg
                fill="none"
                height="100%"
                viewBox="0 0 20 20"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M11.5858 10L7.29289 5.70711C6.90237 5.31658 6.90237 4.68342 7.29289 4.29289C7.68342 3.90237 8.31658 3.90237 8.70711 4.29289L13.7071 9.29289C14.0976 9.68342 14.0976 10.3166 13.7071 10.7071L8.70711 15.7071C8.31658 16.0976 7.68342 16.0976 7.29289 15.7071C6.90237 15.3166 6.90237 14.6834 7.29289 14.2929L11.5858 10Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  stroke="none"
                />
              </svg>
            </div>
          </div>
          <div
            class="title"
          >
            <div
              class="root"
            >
              <div
                class="userInfo"
              >
                <div
                  class="wordAvatar orange-green"
                >
                  ИИ
                </div>
                <div
                  class="userName"
                  title="Иванов Иван"
                >
                  Иванов Иван
                </div>
              </div>
              <div
                class="projects"
              >
                <div
                  class="tag"
                >
                  Smart
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="header"
        >
          <div
            class="arrow"
          >
            <div
              class="root"
            >
              <svg
                fill="none"
                height="100%"
                viewBox="0 0 20 20"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M11.5858 10L7.29289 5.70711C6.90237 5.31658 6.90237 4.68342 7.29289 4.29289C7.68342 3.90237 8.31658 3.90237 8.70711 4.29289L13.7071 9.29289C14.0976 9.68342 14.0976 10.3166 13.7071 10.7071L8.70711 15.7071C8.31658 16.0976 7.68342 16.0976 7.29289 15.7071C6.90237 15.3166 6.90237 14.6834 7.29289 14.2929L11.5858 10Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  stroke="none"
                />
              </svg>
            </div>
          </div>
          <div
            class="title"
          >
            <div
              class="root"
            >
              <div
                class="userInfo"
              >
                <div
                  class="wordAvatar orange-yellow"
                >
                  ПВ
                </div>
                <div
                  class="userName"
                  title="Петров Василий"
                >
                  Петров Василий
                </div>
              </div>
              <div
                class="projects"
              >
                <div
                  class="tag"
                >
                  Smart
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="header"
        >
          <div
            class="arrow"
          >
            <div
              class="root"
            >
              <svg
                fill="none"
                height="100%"
                viewBox="0 0 20 20"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M11.5858 10L7.29289 5.70711C6.90237 5.31658 6.90237 4.68342 7.29289 4.29289C7.68342 3.90237 8.31658 3.90237 8.70711 4.29289L13.7071 9.29289C14.0976 9.68342 14.0976 10.3166 13.7071 10.7071L8.70711 15.7071C8.31658 16.0976 7.68342 16.0976 7.29289 15.7071C6.90237 15.3166 6.90237 14.6834 7.29289 14.2929L11.5858 10Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  stroke="none"
                />
              </svg>
            </div>
          </div>
          <div
            class="title"
          >
            <div
              class="root"
            >
              <div
                class="userInfo"
              >
                <div
                  class="wordAvatar orange-green"
                >
                  ИИ
                </div>
                <div
                  class="userName"
                  title="Иванов Иван"
                >
                  Иванов Иван
                </div>
              </div>
              <div
                class="projects"
              >
                <div
                  class="tag"
                >
                  Smart
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="header"
        >
          <div
            class="arrow"
          >
            <div
              class="root"
            >
              <svg
                fill="none"
                height="100%"
                viewBox="0 0 20 20"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M11.5858 10L7.29289 5.70711C6.90237 5.31658 6.90237 4.68342 7.29289 4.29289C7.68342 3.90237 8.31658 3.90237 8.70711 4.29289L13.7071 9.29289C14.0976 9.68342 14.0976 10.3166 13.7071 10.7071L8.70711 15.7071C8.31658 16.0976 7.68342 16.0976 7.29289 15.7071C6.90237 15.3166 6.90237 14.6834 7.29289 14.2929L11.5858 10Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  stroke="none"
                />
              </svg>
            </div>
          </div>
          <div
            class="title"
          >
            <div
              class="root"
            >
              <div
                class="userInfo"
              >
                <div
                  class="wordAvatar orange-yellow"
                >
                  ПВ
                </div>
                <div
                  class="userName"
                  title="Петров Василий"
                >
                  Петров Василий
                </div>
              </div>
              <div
                class="projects"
              >
                <div
                  class="tag"
                >
                  Smart
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="header"
        >
          <div
            class="arrow"
          >
            <div
              class="root"
            >
              <svg
                fill="none"
                height="100%"
                viewBox="0 0 20 20"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M11.5858 10L7.29289 5.70711C6.90237 5.31658 6.90237 4.68342 7.29289 4.29289C7.68342 3.90237 8.31658 3.90237 8.70711 4.29289L13.7071 9.29289C14.0976 9.68342 14.0976 10.3166 13.7071 10.7071L8.70711 15.7071C8.31658 16.0976 7.68342 16.0976 7.29289 15.7071C6.90237 15.3166 6.90237 14.6834 7.29289 14.2929L11.5858 10Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  stroke="none"
                />
              </svg>
            </div>
          </div>
          <div
            class="title"
          >
            <div
              class="root"
            >
              <div
                class="userInfo"
              >
                <div
                  class="wordAvatar orange-green"
                >
                  ИИ
                </div>
                <div
                  class="userName"
                  title="Иванов Иван"
                >
                  Иванов Иван
                </div>
              </div>
              <div
                class="projects"
              >
                <div
                  class="tag"
                >
                  Smart
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="header"
        >
          <div
            class="arrow"
          >
            <div
              class="root"
            >
              <svg
                fill="none"
                height="100%"
                viewBox="0 0 20 20"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M11.5858 10L7.29289 5.70711C6.90237 5.31658 6.90237 4.68342 7.29289 4.29289C7.68342 3.90237 8.31658 3.90237 8.70711 4.29289L13.7071 9.29289C14.0976 9.68342 14.0976 10.3166 13.7071 10.7071L8.70711 15.7071C8.31658 16.0976 7.68342 16.0976 7.29289 15.7071C6.90237 15.3166 6.90237 14.6834 7.29289 14.2929L11.5858 10Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  stroke="none"
                />
              </svg>
            </div>
          </div>
          <div
            class="title"
          >
            <div
              class="root"
            >
              <div
                class="userInfo"
              >
                <div
                  class="wordAvatar orange-yellow"
                >
                  ПВ
                </div>
                <div
                  class="userName"
                  title="Петров Василий"
                >
                  Петров Василий
                </div>
              </div>
              <div
                class="projects"
              >
                <div
                  class="tag"
                >
                  Smart
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="header"
        >
          <div
            class="arrow"
          >
            <div
              class="root"
            >
              <svg
                fill="none"
                height="100%"
                viewBox="0 0 20 20"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M11.5858 10L7.29289 5.70711C6.90237 5.31658 6.90237 4.68342 7.29289 4.29289C7.68342 3.90237 8.31658 3.90237 8.70711 4.29289L13.7071 9.29289C14.0976 9.68342 14.0976 10.3166 13.7071 10.7071L8.70711 15.7071C8.31658 16.0976 7.68342 16.0976 7.29289 15.7071C6.90237 15.3166 6.90237 14.6834 7.29289 14.2929L11.5858 10Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  stroke="none"
                />
              </svg>
            </div>
          </div>
          <div
            class="title"
          >
            <div
              class="root"
            >
              <div
                class="userInfo"
              >
                <div
                  class="wordAvatar orange-yellow"
                >
                  КК
                </div>
                <div
                  class="userName"
                  title="Кент Кларк"
                >
                  Кент Кларк
                </div>
              </div>
              <div
                class="projects"
              >
                <div
                  class="tag"
                >
                  Smart
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="header"
        >
          <div
            class="arrow"
          >
            <div
              class="root"
            >
              <svg
                fill="none"
                height="100%"
                viewBox="0 0 20 20"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M11.5858 10L7.29289 5.70711C6.90237 5.31658 6.90237 4.68342 7.29289 4.29289C7.68342 3.90237 8.31658 3.90237 8.70711 4.29289L13.7071 9.29289C14.0976 9.68342 14.0976 10.3166 13.7071 10.7071L8.70711 15.7071C8.31658 16.0976 7.68342 16.0976 7.29289 15.7071C6.90237 15.3166 6.90237 14.6834 7.29289 14.2929L11.5858 10Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  stroke="none"
                />
              </svg>
            </div>
          </div>
          <div
            class="title"
          >
            <div
              class="root"
            >
              <div
                class="userInfo"
              >
                <div
                  class="wordAvatar orange-yellow"
                >
                  КК
                </div>
                <div
                  class="userName"
                  title="Кент Кларк"
                >
                  Кент Кларк
                </div>
              </div>
              <div
                class="projects"
              >
                <div
                  class="tag"
                >
                  Smart
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="header"
        >
          <div
            class="arrow"
          >
            <div
              class="root"
            >
              <svg
                fill="none"
                height="100%"
                viewBox="0 0 20 20"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M11.5858 10L7.29289 5.70711C6.90237 5.31658 6.90237 4.68342 7.29289 4.29289C7.68342 3.90237 8.31658 3.90237 8.70711 4.29289L13.7071 9.29289C14.0976 9.68342 14.0976 10.3166 13.7071 10.7071L8.70711 15.7071C8.31658 16.0976 7.68342 16.0976 7.29289 15.7071C6.90237 15.3166 6.90237 14.6834 7.29289 14.2929L11.5858 10Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  stroke="none"
                />
              </svg>
            </div>
          </div>
          <div
            class="title"
          >
            <div
              class="root"
            >
              <div
                class="userInfo"
              >
                <div
                  class="wordAvatar orange-yellow"
                >
                  КК
                </div>
                <div
                  class="userName"
                  title="Кент Кларк"
                >
                  Кент Кларк
                </div>
              </div>
              <div
                class="projects"
              >
                <div
                  class="tag"
                >
                  Smart
                </div>
                <div
                  class="tag"
                >
                  NBA
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
      >
        <div
          class="header"
        >
          <div
            class="arrow"
          >
            <div
              class="root"
            >
              <svg
                fill="none"
                height="100%"
                viewBox="0 0 20 20"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M11.5858 10L7.29289 5.70711C6.90237 5.31658 6.90237 4.68342 7.29289 4.29289C7.68342 3.90237 8.31658 3.90237 8.70711 4.29289L13.7071 9.29289C14.0976 9.68342 14.0976 10.3166 13.7071 10.7071L8.70711 15.7071C8.31658 16.0976 7.68342 16.0976 7.29289 15.7071C6.90237 15.3166 6.90237 14.6834 7.29289 14.2929L11.5858 10Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  stroke="none"
                />
              </svg>
            </div>
          </div>
          <div
            class="title"
          >
            <div
              class="root"
            >
              <div
                class="userInfo"
              >
                <div
                  class="wordAvatar cyan-blue"
                >
                  СА
                </div>
                <div
                  class="userName"
                  title="Сергей Антонов"
                >
                  Сергей Антонов
                </div>
              </div>
              <div
                class="projects"
              >
                <div
                  class="tag"
                >
                  NBA
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
